Bottom: e51e4dd8190256aff7741f30ffac0c43e9f489eb
Top:    0604b949928224cf6430787c44e9bce0630c3744
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-12-27 18:09:33 +0000

Play with grid snapping heuristics a little

Require the pointer to be inside the bounding box of a pin / pad / via before snapping.
This makse it MUCH easier to place vias in a regular grid spaced at the PCB grid spacing.


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 71edaca..9cece49 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -994,7 +994,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PAD_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1051,7 +1051,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PIN_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1069,7 +1069,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (VIA_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid snapping vias to any other vias */
   if (Settings.Mode == MOVE_MODE &&
